processing/data_collection/gazette/spiders/sp_vinhedo.py

```diff
@@ -0,0 +1,56 @@
+from datetime import date, datetime
+import re
+
+from dateparser import parse
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpVinhedoSpider(BaseGazetteSpider):
+
+    # define class variables
+    allowed_domains = ['vinhedo.sp.gov.br']
+    name = 'sp_vinhedo'
+    start_urls = [
+        'http://vinhedo.sp.gov.br/arquivos/?wpdmc=boletim-municipal'
+    ]
+    start_date = date(2010, 12, 9)
+    TERRITORY_ID = '3556701'
+
+    # parse gazette pages dynamically
+    def parse(self, response):
+        pages = response.xpath('//*[@class="page-numbers"]//@href').getall()
```

Comment:
**suggestion**: Scrapy take care of not sending duplicated requests more than once, so you could simply send the page links available on the current page. When we are in a next page, new page links will appear and they will be correctly handled.

When searching for class attribute, give preference for CSS Selectors.

```
for page_url in response.css("a.page-numbers::attr(href)").getall():
    yield scrapy.Request(response.urljoin(page_url), self.parse_page)
```